src/render.tsx

```diff
@@ -1,40 +1,59 @@
 import TestRenderer from 'react-test-renderer';
 import type { ReactTestInstance, ReactTestRenderer } from 'react-test-renderer';
 import * as React from 'react';
+import { Profiler } from 'react';
 import act from './act';
 import { addToCleanupQueue } from './cleanup';
 import debugShallow from './helpers/debugShallow';
 import debugDeep from './helpers/debugDeep';
 import { getQueriesForElement } from './within';
-import { setRenderResult } from './screen';
+import { setRenderResult, screen } from './screen';
+import { assertStringsWithinText } from './helpers/assertStringsWithinText';
 
 type Options = {
   wrapper?: React.ComponentType<any>;
   createNodeMock?: (element: React.ReactElement) => any;
+  experimentalValidateStringsRenderedInText?: boolean;
 };
+
 type TestRendererOptions = {
   createNodeMock: (element: React.ReactElement) => any;
 };
 
 export type RenderResult = ReturnType<typeof render>;
 
+type RenderParams<T> = Options & {
+  component: React.ReactElement<T>;
+  internalWrap?: (innerElement: React.ReactElement) => React.ReactElement;
+};
+
 /**
  * Renders test component deeply using react-test-renderer and exposes helpers
  * to assert on the output.
  */
-export default function render<T>(
-  component: React.ReactElement<T>,
-  { wrapper: Wrapper, createNodeMock }: Options = {}
-) {
-  const wrap = (innerElement: React.ReactElement) =>
-    Wrapper ? <Wrapper>{innerElement}</Wrapper> : innerElement;
+export function render<T>({
```

Comment:
I agree the exports are confusing, I initially didn't want to change the public api of the render function but since there is now an option it's not an issue anymore, I changed it so that there is a single exported function and extracted what was shared in another function, there is still a bit of duplicated code but not that much and it is indeed way clearer imo